File: experimental/workflow_mt/consumer/rust/src/consumer/broadcasts/groupb_groupc_structa.rs
use super::{protocol, Consumer, Context};
use crate::stat::Alias;
use clibri::client;

pub async fn handler<E: client::Error>(
    event: protocol::GroupB::GroupC::StructA,
    context: &mut Context,
    consumer: Consumer<E>,
) {
    context.inc_stat(Alias::GroupBGroupCStructA);
}

File: tests/workflow/producer/typescript (copy)/src/producer/implementation/responses/groupb.groupc.structa.ts
import { Producer, Identification, Filter, Context, Protocol } from "./index";
import { response } from "../../responses/groupb.groupc.structa";

export class Response {
    private _response!: Protocol.GroupB.GroupC.StructB | Protocol.GroupA.StructB;

    constructor(res: Protocol.GroupB.GroupC.StructB | Protocol.GroupA.StructB) {
        this._response = res;
    }

    public pack(sequence: number, uuid: string): ArrayBufferLike {
        return this._response.pack(sequence, uuid);
    }
}

export function handler(
    request: Protocol.GroupB.GroupC.StructA,
    consumer: Identification,
    filter: Filter,
    context: Context,
    producer: Producer,
    sequence: number
): Promise<void> {
    return response(request, consumer, filter, context, producer).then(
        (res) => {
            return producer.send(
                consumer.uuid(),
                res.pack(sequence, consumer.uuid())
            );
        }
    );
}